pkg/kubernetes/token_exchange.go

```diff
@@ -0,0 +1,81 @@
+package kubernetes
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/containers/kubernetes-mcp-server/pkg/config"
+	"github.com/containers/kubernetes-mcp-server/pkg/tokenexchange"
+	"github.com/coreos/go-oidc/v3/oidc"
+	"golang.org/x/oauth2"
+	"k8s.io/klog/v2"
+)
+
+func ExchangeTokenInContext(
```

Comment:
I haven't looked into this, but it seems like having this in the MCP layer has not much sense.

As I see it the HTTP layer sets the Bearer token header that gets propagated to the MCP layer through the context.

Then in the MCP layer we perform the token exchange by invoking provider methods to replace the Bearer token header with the exchanged token.

Then we call GetDerivedKubernetes on the provider to retrieve the applicable Kubernets client-set.

IMO at this point, the MCP layer could just call GetDerived and don't have to deal with auth-related tasks that have nothing to do with the MCP domain.

Reply:
the reason why I added this into the MCP layer is that the token exchange relies on MCP tool call params (in particular the target for the tool). This means that we need to parse the MCP tool call request to extract the target before we can know how to do the token exchange.

If you prefer, we can move this back to the http layer, this will just require us manually parsing the tool call requests (at least the parameter) in the HTTP layer